### PR TITLE
Force cookies with SameSite=None to also include Secure

### PR DESCRIPTION
**Pull request by samouri**
Fixes https://github.com/ampproject/amphtml/issues/25280

cc @choumx @jridgewell

### CODE REVIEW

**Comment by samouri** [comment]
@jridgewell, I've updated `setCookie` to take in a new optional `secure` flag. PTAL

File: src/cookies.js [hunk 1]
```diff
@@ -181,8 +183,9 @@ export function getHighestAvailableDomain(win) {
  * @param {time} expirationTime
  * @param {string|undefined} domain
  * @param {!SameSite=} sameSite
```
Comment (jridgewell) [inline src/cookies.js]: Nit: This signature means that you can either do `foo(sameSite)` or `foo()`. It should be `{!SameSite|undefined}`

Comment (samouri) [reply src/cookies.js]: After reading through [gcc types](https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System), I'm guessing this means that I should also change `{boolean}` --> `{boolean=}`.

Comment (jridgewell) [reply src/cookies.js]: No, we'll always call this function with the correct number of arguments, so `=` is incorrect.

Comment (samouri) [reply src/cookies.js]: Sure, I'll switch it back

File: src/cookies.js [hunk 2]
```diff
@@ -183,7 +183,7 @@ export function getHighestAvailableDomain(win) {
  * @param {time} expirationTime
  * @param {string|undefined} domain
  * @param {!SameSite|undefined} sameSite
- * @param {boolean=} secure
+ * @param {boolean} secure
```
Comment (jridgewell) [inline src/cookies.js]: This can be undefined, too.

Comment (samouri) [reply src/cookies.js]: Lines 157 and 160 call `trySetCookie` without the last two args :(

**Pull request merged**